azure/scope/cluster.go

```diff
@@ -1004,3 +1011,11 @@ func (s *ClusterScope) TagsSpecs() []azure.TagsSpec {
 		},
 	}
 }
+
+func buildServiceEndpointList(subnetServiceEndpoints infrav1.ServiceEndpoints) []infrav1.ServiceEndpointSpec {
```

Comment:
I'm not sure why we need this extra func here, isn't `infrav1.ServiceEndpoints` equivalent to `[]infrav1.ServiceEndpointSpec`?

Reply:
Heh, good point. This originally started because of a field being a pointer (see the subnet/spec.go) but I changed that an never came back and refactored this code. I'll remove this.